calcom/cal.com
# chore: i18n cleanup script

## What does this PR do?



- Fixes #15623
- /claim #15623
- 
## Mandatory Tasks (DO NOT REMOVE)

- [ x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [ x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

Make sure all the correct unused keys are being removed. Also, make sure to add any keys that should not be removed to the KEYS_TO_NOT_REMOVE array.

I initially considered two approaches. The first was to use glob to return a string of each component and then match regex patterns for i18n, t(), and other patterns. The second approach involved parsing the Abstract Syntax Tree (AST) and looking for these patterns to remove. But, I noticed that some parts of the code weren't using these patterns at all, so I decided to take a different route. Now, I'm first getting the keys from the common files and then searching for them in the string of the component that glob provides and removing them if they don't exist.
